backend/src/main/java/outcobra/server/model/Teacher.java

```diff
@@ -0,0 +1,28 @@
+package outcobra.server.model;
+
+import java.util.List;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+
+import lombok.*;
+
+@Data
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+public class Teacher {
+    @Id
+    private long id;
```

Comment:
We usually use `Long` (boxed) here because that prevents some problems with Hibernate.